lib/rspec/mocks/example_methods.rb

```diff
@@ -201,6 +201,14 @@ def allow_message_expectations_on_nil
         RSpec::Mocks.space.proxy_for(nil).warn_about_expectations = false
       end
 
+      # Raise an error when setting expectations on nil.
+      #
+      # By default warning messages are issued when expectations are set on
+      # nil.  This is to force test failures when an expectation is set on nil.
+      def disallow_message_expectations_on_nil
```

Comment:
I think `allow_message_expectations_on_nil` originated in RSpec 1 (before rspec-mocks had its own config) and it just never got moved to global config once we added that.

I'm thinking a single config option (`allow_message_expectations_on_nil`) can suffice. If not set (e.g. `nil`) it will warn as it normally does.  If set to `true` it silences the warning.  If set to `false` it will raise an error instead of warning.

For backwards compatibility we'll have to keep the `allow_message_expectations_on_nil` method for now.